CMakeLists.txt

```diff
@@ -312,7 +312,9 @@ endif()
 # Benchmarks and tests at some places are coupled which is not great. See
 # velox/vector/CMakeLists.txt. TODO: Decouple.
 set(VELOX_DISABLE_GOOGLETEST OFF)
-if(NOT VELOX_BUILD_TESTING AND NOT VELOX_ENABLE_BENCHMARKS_BASIC)
+if(NOT VELOX_BUILD_TEST_UTILS
```

Comment:
There might be other projects of which Velox is a submodule and it seems unnecessary to build test utils etc when you just need gtest as a dependency. How about we have another flag say VELOX_LINK_GTEST which if set wont trigger the section below. It also makes it evident why its set.

Reply:
Ok , sounds good.